pkg/controllers/v1alpha2/mpi_job_controller.go

```diff
@@ -1100,6 +1103,23 @@ func newWorker(mpiJob *kubeflow.MPIJob, desiredReplicas int32) *appsv1.StatefulS
 		},
 	})
 
+	// add SchedulerName to podSpec
+	if enableGangScheduling {
+		if podSpec.Spec.SchedulerName != "" && podSpec.Spec.SchedulerName != gangSchedulerName {
+			errMsg := fmt.Sprintf(
+				"%s scheduler is specified when gang-scheduling is enabled and it will not be overwritten", podSpec.Spec.SchedulerName)
+			glog.Warning(errMsg)
+		} else {
+			podSpec.Spec.SchedulerName = gangSchedulerName
+		}
+
+		if podSpec.Annotations == nil {
+			podSpec.Annotations = map[string]string{}
+		}
+		// we create the podGroup with the same name as the mpijob
+		podSpec.Annotations["scheduling.k8s.io/group-name"] = mpiJob.Name
```

Comment:
use a const here

Reply:
@fisherxu Done

Reply:
CI seems failed, let's fix it and get merged.

Reply:
CI is green now, PTAL